### PR TITLE
Add max-pods e2e test

### PR DESCRIPTION
**Pull request by gmarek**

### CODE REVIEW

**Comment by davidopp** [comment]
Tests PR #5547

File: test/e2e/max_pods.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+			expectNoError(err)
+		}
+
+		By(fmt.Sprintf("Destroying namespace for this suite %v", ns))
+		if err := c.Namespaces().Delete(ns); err != nil {
+			Failf("Couldn't delete ns %s", err)
+		}
+	})
+
+	It("Validates MaxPods limit number of pods that are allowed to run.", func() {
```
Comment (davidopp) [inline test/e2e/max_pods.go]: Maybe add a comment that this test can't run in parallel with other e2e's, and also you expect the cluster add-on pods (which should therefore be the only pods running other than the ones you create in the test) to stay stable.

**Comment by davidopp** [comment]
Could you look at the events for the Pod to make sure you see a PodFitsResourcesNode failure (like
```
"failedScheduling    Error scheduling: failed to find fit for pod, Node xxx: PodFitsResourcesNode ..."
```
or whatever it is these days)? This would give more assurance that the Pod was actually rejected by the scheduler rather than we just didn't wait long enough.

**Comment by gmarek** [comment]
Yes, I can look into events. I was looking for some 'why pending' in pod status, but it seems that it was moved somewhere else...

File: test/e2e/max_pods.go (hunk 1 above)
Comment (gmarek) [inline test/e2e/max_pods.go]: Done.

**Comment by gmarek** [comment]
PTAL

**Comment by davidopp** [comment]
@k8s-bot test this please

**Comment by davidopp** [comment]
I talked with @quinton-hoole and in order to make this test not run in parallel with other tests, you need to add a modification to this line of hack/jenkins/e2e.sh to your PR:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/hack/jenkins/e2e.sh#L74

**Comment by davidopp** [comment]
LGTM

**Comment by wojtek-t** [comment]
@davidopp - can you please apply "ok-to-merge" label?

**Comment by davidopp** [comment]
Done

**Comment by wojtek-t** [comment]
Thanks - will merge tomorrow during our business hours.

**Pull request merged**